bndtools.core/src/bndtools/dnd/gav/GAVIPageListener.java

```diff
@@ -0,0 +1,112 @@
+package bndtools.dnd.gav;
+
+import static org.eclipse.swt.dnd.DND.DROP_COPY;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.swt.custom.StyledText;
+import org.eclipse.swt.dnd.TextTransfer;
+import org.eclipse.swt.dnd.Transfer;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.IFileEditorInput;
+import org.eclipse.ui.IPartListener;
+import org.eclipse.ui.IStorageEditorInput;
+import org.eclipse.ui.IWorkbenchPart;
+import org.eclipse.ui.dnd.IDragAndDropService;
+import org.eclipse.ui.part.EditorPart;
+
+import aQute.lib.exceptions.Exceptions;
+import bndtools.editor.BndEditor;
+
+public class GAVIPageListener implements IPartListener {
+
+	private final Transfer[] transfers = new Transfer[] {
+		TextTransfer.getInstance()
+	};
+
+	@Override
+	public void partOpened(IWorkbenchPart part) {
+		IDragAndDropService dndService = part.getSite()
+			.getService(IDragAndDropService.class);
+
+		if (dndService == null) {
+			return;
+		}
+
+		if (part instanceof BndEditor) {
+			part = ((BndEditor) part).getSourcePage();
```

Comment:
Perhaps we should modify the `BndEditor` so that you don't have to do this.  The `BndEditor` could override the `getAdapter` method and return the styledText where someone queries for `Control.class`

Reply:
fixed!